tests/persistance/metadataWriter.spec.ts

```diff
@@ -0,0 +1,20 @@
+import MetadataWriter from "../../src/persistance/metadataWriter";
+import Node from "../../src/models/node";
+import { META_DATA_LOCATION } from "../../src/constants";
+import fs from "fs";
+
+jest.mock("fs");
+
+describe("Persistent metadata", () => {
+  it("should generate metadata for a node", () => {
+    const node = new Node(32);
+    const metaData = node.metadata();
+    const metadataWriter = new MetadataWriter(fs);
+    metadataWriter.writeMetadata(metaData);
+    expect(fs.writeFileSync).toHaveBeenCalledWith(
```

Comment:
It's generally a good practice to hardcode data in test files.
So, instead of:
```
expect(fs.writeFileSync).toHaveBeenCalledWith(
  META_DATA_LOCATION + "/" + metaData.type + ".dat",
  JSON.stringify(metaData.data),
  "utf-8"
);
```
...maybe use:

```
expect(fs.writeFileSync).toHaveBeenCalledWith(
  "/metadata/node/32.dat",
  "{}"),
  "utf-8"
);
```

That way, tests serve as documentation to code and also, if some refactor messes up with the file path, it would be caught.

Reply:
I can't do that. Because I am getting the __dirname dynamically. For the test and the src the __dirname won't be the same. Hence I was doing that.